src/controller_examples/rabbitmq_controller/spec/reconciler.rs

```diff
@@ -675,22 +675,29 @@ pub open spec fn make_stateful_set(rabbitmq: RabbitmqClusterView, config_map_rv:
             )
             .set_spec(make_rabbitmq_pod_spec(rabbitmq))
         )
-        .set_volume_claim_templates(seq![
-            PersistentVolumeClaimView::default()
-                .set_metadata(ObjectMetaView::default()
-                    .set_name(new_strlit("persistence")@)
-                    .set_namespace(namespace)
-                    .set_labels(labels)
-                )
-                .set_spec(PersistentVolumeClaimSpecView::default()
-                    .set_access_modes(seq![new_strlit("ReadWriteOnce")@])
-                    .set_resources(ResourceRequirementsView::default()
-                        .set_requests(Map::empty()
-                            .insert(new_strlit("storage")@, new_strlit("10Gi")@)
+        .set_volume_claim_templates({
+            if rabbitmq.spec.persistence.storage == new_strlit("0Gi")@ {
+                seq![]
+            } else {
+                seq![
+                    PersistentVolumeClaimView::default()
+                        .set_metadata(ObjectMetaView::default()
+                            .set_name(new_strlit("persistence")@)
+                            .set_namespace(namespace)
+                            .set_labels(labels)
                         )
-                    )
-                )
-        ])
+                        .set_spec(PersistentVolumeClaimSpecView::default()
+                            .set_access_modes(seq![new_strlit("ReadWriteOnce")@])
+                            .set_resources(ResourceRequirementsView::default()
+                                .set_requests(Map::empty()
+                                    .insert(new_strlit("storage")@, rabbitmq.spec.persistence.storage)
+                                )
+                            )
+                            .set_storage_class_name(rabbitmq.spec.persistence.storage_class_name)
```

Comment:
For spec code, we actually don't have to construct the object by calling `set_` or `overwrite_`. We can just write down the struct literal like here: https://github.com/vmware-research/verifiable-controllers/blob/main/src/controller_examples/zookeeper_controller/spec/reconciler.rs#L1080, and it is even clearer since we don't need to know what is behind these set functions. I am planning to rewrite the spec in this way but we don't have to do it in one PR.